### ISSUE TITLE
Redesign of file viewer 

### ISSUE THREAD

**Issue by dbranchini** [sul-dlss/purl#801]
The need for a redesign started with an accessibility issue - in the current design, each item contains two links, and both links go to the same place. Using a screen reader, the user hears each link twice, which might cause confusion and definitely results in audio clutter. See the screenshot in the comment below. Also, because each item contains two separate links, this adds extra tabs to the keyboard navigation to get to an item further down in the list or to the "download all" and share features. Amy Hodge and I then discussed that the call to action in the current design isn't apparent, and to add to it, Justin Coyne reached out to ask about changing the current design slightly because the aesthetic looks off balance. There's a tiny icon in a large square next to each item. I'd also suggest that the iframe window is small vertically, and we don't use the space horizontally very well. Finally, I thought a familiar pattern, such as Mac's Finder, Window's File Explorer or Google Drive, could inspire our redesign since, dare I say, every user will be familiar with one of those tools.

Given the inspiration outlined above, I based my redesign on that, and reviewed it with Amy, Justin, and Andrew Berger. During that conversation, we also discussed the following requirements:

- Preview isn't required. Per Andrew, it's only available on a couple of items, so it shouldn't be propagated to the new design.
- The file viewer is the only viewer with a search feature.
- Download all isn't always available. If it's above a certain size, the user must contact us to obtain the items. 
- There's essentially no character limit on the description. It can be up to 65,000 + characters. The current design limits what is displayed. 
- File numbers aren't required. It's on the current design, but no one knows how they're used and/or if they're used.  

I'm sharing my redesign as a flat file for now. Please provide feedback. Additional work is required to be sure that Alan's designs on the media viewer and my designs on the file viewer are consistent in any shared features, such as the "share this item" feature. Once the redesign is solidified, I'll share the Figma for InVision file. 

Desktop
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMm0yQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--d4418d5f39224b156190e4f92a95e3cd1703105e/Screenshot%202023-11-08%20at%209.54.53%20AM.png)

Mobile
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMmUyQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--ea1733f9d8c3735712b6d47b84fb78745c2092fa/Screenshot%202023-11-08%20at%209.55.42%20AM.png)

**Comment by dbranchini** [comment]
Here are the screenshots of the current design with the duplication of links. 

<img width="1181" alt="Screenshot 2023-11-08 at 9 59 12 AM" src="https://github.com/sul-dlss/purl/assets/6816624/b8337753-c498-49d5-9d74-13acd9265db6">

<img width="1198" alt="Screenshot 2023-11-08 at 9 57 51 AM" src="https://github.com/sul-dlss/purl/assets/6816624/7420e996-523f-466d-82f9-775e4b1dbcdc">

**Comment by dbranchini** [comment]
I was asked to look at Alan's redesign of the media viewer and to think about consistency across the various viewers. His design was inspired by the **Mirador image viewer.** The file viewer design is inspired by Alan's work and Mirador and seeks to be more consistent with those designs while considering the needs of users for file viewing activities specifically. The header for the file viewer has a similar look and feel, but the specific tool and feature options are different than the image and media viewers. 

Here's the Figma prototype - https://www.figma.com/proto/LVP13xZ0HhrxAPHCnoMVW1/Embedded-File-Viewer?page-id=0%3A1&type=design&node-id=202-1102&viewport=-589%2C-316%2C0.5&t=N0cHKqEDLgwovQDv-1&scaling=min-zoom&starting-point-node-id=188%3A795&show-proto-sidebar=1&mode=design

See later comments for expected interactions.

**Comment by dbranchini** [comment]
Mobile idea - https://www.figma.com/proto/EsaColfZL9Yny67db6Nnfp/File-Viewer---Mobile?page-id=0%3A1&type=design&node-id=45-967&viewport=-445%2C1916%2C1&t=AhAUR2E9oR3KYend-1&scaling=min-zoom&starting-point-node-id=45%3A966&mode=design. 

If we have the capacity to do a mobile version, I can complete this prototype. None of our viewers seem to be 100% mobile compliant.

**Comment by justinlittman** [comment]
@dbranchini Can you please provide a design showing multiple levels of hierarchy? That's the sort of thing that always gets us tied into knots with these sorts of views.

**Comment by dbranchini** [comment]
@justinlittman , can you point me to an example in our current viewer?

**Comment by amyehodge** [comment]
@dbranchini @justinlittman I think this one will work for you: https://purl.stanford.edu/fg478vy8624

**Comment by dbranchini** [comment]
@justinlittman @amyehodge here's the [prototype of the viewer with multiple hierarchies](https://www.figma.com/proto/LVP13xZ0HhrxAPHCnoMVW1/Embedded-File-Viewer?page-id=0%3A1&type=design&node-id=316-1113&viewport=-2703%2C-153%2C0.5&t=FKcfJUrDHWARGZAl-1&scaling=min-zoom&starting-point-node-id=316%3A1113&mode=design)

**Comment by justinlittman** [comment]
@dbranchini How is the "every other row is gray" supposed to work with expanding / collapsing? Won't this end up with odd graying depending on the number of hidden children?

**Comment by justinlittman** [comment]
This seems like an instance of https://www.w3.org/WAI/ARIA/apg/patterns/treegrid/

**Comment by dbranchini** [comment]
@justinlittman, great find! I love that site for accessible patterns. Yes, I agree that the design is an instance of a treegrid and should follow the keyboard navigation guidelines described here. 

I see what you're saying about the pattern of gray/white. I'm looking at Finder (Mac), and it dynamically and subtly changes them when you open/close folders. Is that possible?

**Comment by justinlittman** [comment]
Re patterns of gray/white: It depends. If this can be done with CSS, it is feasible. (For example, using a variation of the CSS used for https://getbootstrap.com/docs/5.3/content/tables/#striped-rows.) If we have to do it in Javascript, I'd suggest that the complexity isn't worth it. We won't be able to tell until the tree grid is written.

**Comment by dbranchini** [comment]
Fair enough; if it can't be done without a significant time sync, then I added a version with lines between rows to the prototype that would also work. It's the first screen or starting point of the prototype.
